Merge 18.1.0.2 beta (includes beta 1 and 2) plus release notes into develop

mokagio opened the pull request:
Includes:

**18.1.0.1**

- Enabled the "Recommend app" feature via feature flag by @dvdchr ([#17065](https://github.com/wordpress-mobile/WordPress-iOS/pull/17065))

**18.1.0.2**

- Improve account closure error messaging by @nikola-milicevic ([#17075](https://github.com/wordpress-mobile/WordPress-iOS/pull/17075)

Nothing to test, if CI Shows green, we're good.

## Regression Notes
1. Potential unintended areas of impact
N.A.


2. What I did to test those areas of impact (or what existing automated tests I relied on)
N.A.

3. What automated tests I added (or what prevented me from doing so)
N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

dvdchr committed:
Enable feature flag for recommend app feature

dvdchr committed:
Update release notes

dvdchr committed:
Merge pull request #17065 from wordpress-mobile/feature/enable-recommend-app

Recommend App: Enable feature flag to public

Commit:
Fix displayed localizable string in case of atomic-site error

mokagio committed:
Update translations

mokagio committed:
Update metadata translations

mokagio committed:
Bump version number

mokagio committed:
Manually track the deleted Jetpack older release notes localizations

There's clearly a bug in my rough implementation of the logic to decide
whether to commit the Jetpack metadata. Likely due to manually checking
the Git commands text output instead of a more appropriate usage of Git
machine readable output like @AliSoftware suggested here:
https://github.com/wordpress-mobile/WordPress-iOS/pull/17047#discussion_r691042001

These would get deleted anyway or updated with the localizations once
ready upon a new beta creation or release finalization, but I thought it
best to track it here from a workflow consistency point of view.

mokagio reviewed (commented):
- [x] Version update in `.xcconfig`
- [x] Updates to the localization files (`.strings` and Fastlane metadata) automatically pulled for the new translations that have already been approved in GlotPress
- [x] Diffs from the PRs that made it into this beta

Notice that _all_ the `release_notes.txt` were deleted because we don't have translations yet—It's only the first day after the GlotPress upload run.

mokagio commented:
_I enabled auto-merge before seeing if the tests passed because it's dinner time for me and I don't want to forget_ 🍝

AliSoftware commented on fastlane/metadata/de-DE/subtitle.txt:
```diff
@@ -1 +1 @@
-Erstelle deine neue Website
+Erstelle eine Website oder ein Blog
```
Just a heads up that this will exceed the 30 char limit. (I remember you fixing it manually last time and asking translators for a fix at the source in some P2 or Slack, but apparently they didn't tackle it yet)

mokagio replied:
Yep. This keeps happening. Your prompt was the push I needed to suggest these 30-characters-compliant translations into GlotPress myself. 🤞

AliSoftware replied:
Cool. Let's merge it as is for now and hope the fixed copy gets accepted in GlotPress by the time you'll do the final release then 👍

AliSoftware commented on fastlane/metadata/fr-FR/subtitle.txt:
```diff
@@ -1 +1 @@
-Construisez un site web
+Construisez un site web ou un blog
```
Ditto.

AliSoftware commented on fastlane/metadata/tr/subtitle.txt:
```diff
@@ -1 +1 @@
-Bir web sitesi oluşturun
+Bir web sitesi veya blog oluşturun
```
Ditto.

AliSoftware replied:
![image](https://user-images.githubusercontent.com/216089/130740116-522c348c-98b3-4199-96a8-8f3bd7796013.png)

AliSoftware reviewed (commented):
I notice that you didn't regenerate the strings via `ios_localize_project` as discussed/suggested in p1629722771248800/1629715763.241600-slack-CC7L49W13 after the need for one of the original string copy to be updated – see also https://wordpress.slack.com/archives/C02RQC4LY/p1629710931010500 ; which means this new string, if it was ready, wouldn't be imported in GlotPress.

Is this intentional (i.e. the discussion wasn't settled and the new string not ready yet) or an oversight?

mokagio committed:
Remove atomic site mention from close account messaging (#17075)

mokagio committed:
Update strings for localization

mokagio committed:
Add editorialized release notes for version 18.1

mokagio committed:
Update metadata strings

mokagio committed:
Update metadata strings

mokagio committed:
Manually fix syntax error in generated AppStoreStrings.po files

Note that we're working on a fix for this:
https://github.com/wordpress-mobile/release-toolkit/pull/297

mokagio committed:
Update translations

mokagio committed:
Bump version number

mokagio commented:
Thanks @AliSoftware. The localizations not being updated was intentional, as I was waiting fro input on the PR that changed one.

Since your comment, I got that info, merged the PR and shipped a new beta. The diff does reflect that.

Can I ask for another review? 😄 🙏 Thanks!

mokagio commented on WordPress/Resources/ar.lproj/Localizable.strings:
```diff
@@ -7035,6 +7094,18 @@ translators: Block name. %s: The localized block name */
+/* Error message displayed when unable to close user account due to unresolved chargebacks. */
+"This user account cannot be closed if there are unresolved chargebacks." = "This user account cannot be closed if there are unresolved chargebacks.";
+
+/* Error message displayed when unable to close user account due to having active atomic site. */
+"This user account cannot be closed while it has active atomic sites." = "This user account cannot be closed while it has active atomic sites.";
```
Notice that this and all the other non-English US localization of this string, match the original version not the new one updated from the change in `WordPress/Classes/ViewRelated/Me/Account Settings/AccountSettingsViewController.swift`.

That's expected because these files were downloaded GlotPress during the `download_localized_strings_and_metadata` step of the `new_beta_release`. Because the change to this string hasn't been merged into `develop` yet, GlotPress still has the original version.

AliSoftware replied:
👌

mokagio reviewed (commented):
- [x] Version update in `.xcconfig`
- [x] Updates to the localization files (`.strings` and Fastlane metadata) automatically pulled for the new translations that have already been approved in GlotPress
- [x] Diffs from the PRs that made it into this beta
- [x] `AppStoreStrings.po` changed with editorialized release notes

AliSoftware commented on WordPress/Classes/ViewRelated/Me/Account Settings/AccountSettingsViewController.swift:
```diff
@@ -288,7 +288,7 @@ private class AccountSettingsController: SettingsController {
                                      comment: "Error message displayed when unable to close user account due to being unauthorized.")
         case "atomic-site":
-            return NSLocalizedString("This user account cannot be closed while it has active atomic sites.",
+            return NSLocalizedString("This user account could not be closed. Please contact us by going to Me > Help & Support screen.",
```
Personal note: I'd have said "can't be closed _automatically_" or something like this, to avoid making the user think that "it won't be possible to close it at all" but instead tell them "you need to contact us so we can do some stuff in our end to close it", which would be less worrying than "can't be closed [at all]" imho 🤷‍♂️

AliSoftware reviewed (approved):
✅ Version bump
✅ strings updated, including the last minute post-code-freeze change
✅ release notes being different for WP and JP and matching the copies suggested by editorial

:shipit:
